test/pipelinerun_test.go

```diff
@@ -213,23 +219,177 @@ func getHelloWorldPipelineWithSingularTask(suffix int, namespace string) *v1alph
+						Type:       v1alpha1.PipelineResourceTypeGit,
+						TargetPath: "brandnewspace",
+					}},
+				},
+				Steps: []corev1.Container{{
+					Name:    "write-stuff",
+					Image:   "ubuntu",
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "echo some stuff > /workspace/brandnewspace/stuff"},
+				}, {
+					Name:    "write-other-stuff",
+					Image:   "ubuntu",
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "echo some other stuff > /workspace/brandnewspace/other-stuff"},
+				}},
+			},
+		}, &v1alpha1.Task{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "check-create-files-exists",
+			},
+			Spec: v1alpha1.TaskSpec{
+				Inputs: &v1alpha1.Inputs{
+					Resources: []v1alpha1.TaskResource{{
+						Name: "workspace",
+						Type: v1alpha1.PipelineResourceTypeGit,
+					}},
+				},
+				Steps: []corev1.Container{{
+					Name:    "list",
+					Image:   "ubuntu",
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "ls /workspace"},
+				}, {
+					Name:    "read-stuff",
+					Image:   "ubuntu",
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "cat /workspace/stuff"},
+				}, {
+					Name:    "write-something",
+					Image:   "ubuntu",
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "echo something to a file > /workspace/something"},
+				}},
+			},
+		}, &v1alpha1.Task{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "check-create-files-exists-2",
+			},
+			Spec: v1alpha1.TaskSpec{
+				Inputs: &v1alpha1.Inputs{
+					Resources: []v1alpha1.TaskResource{{
+						Name: "workspace",
+						Type: v1alpha1.PipelineResourceTypeGit,
+					}},
+				},
+				Steps: []corev1.Container{{
+					Name:    "read-stuff",
+					Image:   "busybox",
+					Command: []string{"cat"},
+					Args:    []string{"/workspace/other-stuff"},
+				}, {
+					Name:    "write-something-else",
+					Image:   "ubuntu",
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "echo something else to a file > /workspace/something-else"},
+				}},
+			},
+		}, &v1alpha1.Task{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "read-files",
+			},
+			Spec: v1alpha1.TaskSpec{
+				Inputs: &v1alpha1.Inputs{
+					Resources: []v1alpha1.TaskResource{{
+						Name:       "workspace",
+						Type:       v1alpha1.PipelineResourceTypeGit,
+						TargetPath: "readingspace",
+					}},
+				},
+				Steps: []corev1.Container{{
+					Name:    "read-stuff",
+					Image:   "ubuntu",
+					Command: []string{"cat"},
+					Args:    []string{"/workspace/readingspace/stuff"},
+				}, {
+					Name:    "read-something",
+					Image:   "ubuntu",
+					Command: []string{"cat"},
+					Args:    []string{"/workspace/readingspace/something"},
+				}, {
+					Name:    "read-something-else",
+					Image:   "ubuntu",
+					Command: []string{"cat"},
+					Args:    []string{"/workspace/readingspace/something-else"},
```

Comment:
> what do you think if we had the tasks themselves assert? i.e. have the step fail if the expected text doesnt exist, e.g. using grep
I like the idea. I will update the tasks to assert contents.

> im a bit confused by what each task and each step in each task is doing - maybe we could use the names of the steps to make it a bit more clear

I tried to name tasks with "stuff" and "something" because the tasks are actually writing the same words. I could update them to pattern you mentioned.